src/Test/Html/Selector.elm

```diff
@@ -192,26 +192,67 @@ tag name =
     Tag name
 
 
-{-| Matches elements that have the given attribute. Any attributes that can
-exist with `String` or `Bool` values like `title` or `disabled` will be matched.
-`Html.Attributes.class` and `Html.Attributes.classList` will work the same as
-`Selectors.classes`. `Html.Attributes.style` will work the same way as
-`Selectors.styles`.
+{-| Matches elements that have the given attribute in a way that makes sense
+given their semantics in `Html`.
+
+
+#### How matching works:
+
+  - `Html.Attributes.class` and `Html.Attributes.classList` will work the same as
+    `Selectors.classes`, matching any element with at least the given classes.
+
+  - `Html.Attributes.style` will work the same way as `Selectors.styles`, matching
+    any element with at least the given style properties.
+
+  - Any other `String` attributes like `title` or `Bool` attributes like
+    `disabled` will match elements with the exact value for those attributes.
+
+  - Any attributes from `Html.Events`, or attributes with values that have types
+    other than `String` or `Bool` will not match anything.
+
+The example below demonstrates usage
```

Comment:
This documentation is way too long to fit nicely on the package site -- better to split it out into the README, and link to it instead for "a fully example"